tox.ini

```diff
@@ -5,23 +5,22 @@
 [tox]
```

Comment:
So tox did not install as part of installing mozanalysis. Is this expected? Are users intended to have their own tox version?

Reply:
That's a good question, maybe we want to add a `dev` optional dependency list?  pip-tools is in the same boat

Reply:
Added optional dev dependencies and update the requirements files to use those

Reply:
Considered adding separate requirements files for dev dependencies but it felt a little too much so I didn't

Reply:
Yeah that's basically it.  It might make sense to just combine it into one because the only real use case for separating them that I can think of is having the CI using `testing` while developers use `dev`, but we're generating the requirements files from `dev` and that's what the CI is using to setup so 🤷

Reply:
I guess the counterpoint is that it does provide a little self-documentation to keep them separate, and because the toml file lets us reference the testing deps in dev we're not introducing any potential for them to get out of sync

Reply:
I'm good to keep them separate. Is updating the CI to use the `testing` requirements in scope for this PR or should we wait on that?

Reply:
I don't think we need to do it at all but I would like to hear Anna or Mike's opinion honestly.  As I see it, tox is installing in the venv that is created by tox, which could cause problems maybe but it doesn't seem to be so 🤷

Reply:
I think installing tox inside tox is probably not going to cause problems, but definitely feels like it would indicate a misconfiguration or misuse somewhere (we shouldn't be running tox inside tox, at the very least).

However, `tox` is only in the `dev` list, which shouldn't be needed for CI, so I think using `testing` inside CI makes sense. I don't think that would be a complicated change, but CI changes are often annoying so if you'd rather tackle it in a follow-up that's fine with me. Also fine to put it on the backlog as it doesn't seem like a high priority or high value change either.

Reply:
So I ended up just putting the dev requirements in their own file.  It was an easy update to `update_deps`, now the CI can use the normal requirements.txt and it won't have tox. let me know what you think.

Reply:
I'm not opposed, but does there have to be so much overlap? It would be nice if the dev one could have *just* the dev deps (but maybe the majority of stuff in both of them is sub-dependencies?).

Does the README need to be updated to account for this change?

Reply:
So I looked through the docs and we can use the `-c` flag so that the `requirements-dev.txt` will be constrained by the `requirements.txt`. It's still overlapping but at least we don't have to worry about conflicts

Reply:
And the documentation should reflect this change, I distinguish between the two files in the relevant sections

Reply:
relevant docs on the c flag: https://github.com/jazzband/pip-tools?tab=readme-ov-file#workflow-for-layered-requirements